--- a/src/gcal-utils.h	2019-05-20 15:03:19.599931086 +0800
+++ b/src/gcal-utils.h	2019-05-20 15:03:24.230730246 +0800
@@ -36,7 +36,6 @@
 
 /* Add autoptr support for libecal */
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (ECalComponent, g_object_unref)
-G_DEFINE_AUTOPTR_CLEANUP_FUNC (ESource, g_object_unref)
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (GWeatherLocation, gweather_location_unref)
 
 GType                icaltime_get_type                           (void)            G_GNUC_CONST;
