--- a/src/libaio.h	2018-10-23 03:59:34.000000000 +0800
+++ b/src/libaio.h	2019-07-07 18:38:28.023878240 +0800
@@ -52,7 +52,7 @@
 
 /* little endian, 32 bits */
 #if defined(__i386__) || (defined(__arm__) && !defined(__ARMEB__)) || \
-    defined(__sh__) || defined(__bfin__) || defined(__MIPSEL__) || \
+    defined(__sh__) || defined(__bfin__) || (defined(__MIPSEL__) && (_MIPS_SIM != _ABI64)) || \
     defined(__cris__) || (defined(__riscv) && __riscv_xlen == 32) || \
     (defined(__GNUC__) && defined(__BYTE_ORDER__) && \
          __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__ && __SIZEOF_LONG__ == 4)
@@ -63,6 +63,7 @@
 /* little endian, 64 bits */
 #elif defined(__ia64__) || defined(__x86_64__) || defined(__alpha__) || \
       (defined(__aarch64__) && defined(__AARCH64EL__)) || \
+      (defined(__MIPSEL__) && (_MIPS_SIM == _ABI64)) || \
       (defined(__riscv) && __riscv_xlen == 64) || \
       (defined(__GNUC__) && defined(__BYTE_ORDER__) && \
           __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__ && __SIZEOF_LONG__ == 8)
@@ -74,6 +75,7 @@
 #elif defined(__powerpc64__) || defined(__s390x__) || \
       (defined(__sparc__) && defined(__arch64__)) || \
       (defined(__aarch64__) && defined(__AARCH64EB__)) || \
+      (defined(__MIPSEB__) && (_MIPS_SIM == _ABI64)) || \
       (defined(__GNUC__) && defined(__BYTE_ORDER__) && \
            __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__ && __SIZEOF_LONG__ == 8)
 #define PADDED(x, y)	unsigned y; x
@@ -83,7 +85,7 @@
 /* big endian, 32 bits */
 #elif defined(__PPC__) || defined(__s390__) || \
       (defined(__arm__) && defined(__ARMEB__)) || \
-      defined(__sparc__) || defined(__MIPSEB__) || defined(__m68k__) || \
+      defined(__sparc__) || (defined(__MIPSEB__) && (_MIPS_SIM != _ABI64)) || defined(__m68k__) || \
       defined(__hppa__) || defined(__frv__) || defined(__avr32__) || \
       (defined(__GNUC__) && defined(__BYTE_ORDER__) && \
            __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__ && __SIZEOF_LONG__ == 4)
