From 8e75e13d9818c5de106cea66087430292dbc26a1 Mon Sep 17 00:00:00 2001
From: WANG Xuerui <git@xen0n.name>
Date: Sun, 24 Sep 2023 01:53:27 +0800
Subject: [PATCH] [lld][LoongArch] Ignore R_LARCH_ALIGN for now

---
 lld/ELF/Arch/LoongArch.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lld/ELF/Arch/LoongArch.cpp b/lld/ELF/Arch/LoongArch.cpp
index 04ddb4682917..faa07acfaaea 100644
--- a/lld/ELF/Arch/LoongArch.cpp
+++ b/lld/ELF/Arch/LoongArch.cpp
@@ -519,6 +519,7 @@ RelExpr LoongArch::getRelExpr(const RelType type, const Symbol &s,
   case R_LARCH_TLS_GD_HI20:
     return R_TLSGD_GOT;
   case R_LARCH_RELAX:
+  case R_LARCH_ALIGN:
     // LoongArch linker relaxation is not implemented yet.
     return R_NONE;
 
@@ -681,6 +682,7 @@ void LoongArch::relocate(uint8_t *loc, const Relocation &rel,
     return;
 
   case R_LARCH_RELAX:
+  case R_LARCH_ALIGN:
     return; // Ignored (for now)
 
   default:
-- 
2.42.0

