--- a/m4/ax_boost_base.m4	2019-07-13 14:59:12.567268752 +0800
+++ b/m4/ax_boost_base.m4	2019-07-13 14:58:37.229268120 +0800
@@ -95,6 +95,9 @@
       x86_64)
         libsubdirs="lib64 libx32 lib lib64"
         ;;
+      mips64)
+        libsubdirs="lib64 lib32 lib lib64"
+        ;;
       ppc64|s390x|sparc64|aarch64|ppc64le)
         libsubdirs="lib64 lib lib64 ppc64le"
         ;;
--- a/configure
+++ b/configure
@@ -20139,6 +20139,9 @@
       x86_64)
         libsubdirs="lib64 libx32 lib lib64"
         ;;
+      mips64)
+        libsubdirs="lib64 lib32 lib lib64"
+        ;;
       ppc64|s390x|sparc64|aarch64|ppc64le)
         libsubdirs="lib64 lib lib64 ppc64le"
         ;;
