From 8f9206979f3131392ad11ccc042a8ff669d18152 Mon Sep 17 00:00:00 2001
From: WANG Xuerui <git@xen0n.name>
Date: Sat, 8 May 2021 16:44:51 +0800
Subject: [PATCH 4/6] allow building without --with-arch=loongarch configure
 arg

---
 gcc/config.gcc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gcc/config.gcc b/gcc/config.gcc
index 3e450dfa203..3668f73ad38 100644
--- a/gcc/config.gcc
+++ b/gcc/config.gcc
@@ -5446,6 +5446,7 @@ case ${target} in
 		fi
 		case ${default_loongarch_arch} in
 		    loongarch) tm_defines="$tm_defines LARCH_ISA_DEFAULT=65" ;;
+		    *) tm_defines="$tm_defines LARCH_ISA_DEFAULT=65" ;;
 		esac
 		case ${default_loongarch_abi} in
 		    lp64)   tm_defines="$tm_defines LARCH_ABI_DEFAULT=ABILP64" ;;
-- 
2.30.1

