diff -pNur mbedtls-mbedtls-2.19.1-orig/include/CMakeLists.txt mbedtls-mbedtls-2.19.1/include/CMakeLists.txt
--- mbedtls-mbedtls-2.19.1-orig/include/CMakeLists.txt   2019-12-20 09:24:46.806987628 +0100
+++ mbedtls-mbedtls-2.19.1/include/CMakeLists.txt        2019-12-20 09:25:51.257199271 +0100
@@ -5,6 +5,9 @@ if(INSTALL_MBEDTLS_HEADERS)
     file(GLOB headers "mbedtls/*.h")
     file(GLOB psa_headers "psa/*.h")
 
+    # Don't overwrite the global config.h file!
+    list(REMOVE_ITEM headers "${CMAKE_CURRENT_SOURCE_DIR}/mbedtls/config.h")
+
     install(FILES ${headers}
         DESTINATION include/mbedtls
         PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ)
