diff -ur a/dlls/winealsa.drv/mmdevdrv.c b/dlls/winealsa.drv/mmdevdrv.c
--- a/dlls/winealsa.drv/mmdevdrv.c	2017-06-15 10:47:39.854223027 +0800
+++ b/dlls/winealsa.drv/mmdevdrv.c	2017-06-15 10:50:52.590214966 +0800
@@ -53,9 +53,9 @@
 
 #define NULL_PTR_ERR MAKE_HRESULT(SEVERITY_ERROR, FACILITY_WIN32, RPC_X_NULL_REF_POINTER)
 
-static const REFERENCE_TIME DefaultPeriod = 100000;
-static const REFERENCE_TIME MinimumPeriod = 50000;
-#define                     EXTRA_SAFE_RT   40000
+static const REFERENCE_TIME DefaultPeriod = 20000;
+static const REFERENCE_TIME MinimumPeriod = 10000;
+#define                     EXTRA_SAFE_RT   5000
 
 struct ACImpl;
 typedef struct ACImpl ACImpl;
diff -ur a/dlls/wineoss.drv/mmdevdrv.c b/dlls/wineoss.drv/mmdevdrv.c
--- a/dlls/wineoss.drv/mmdevdrv.c	2017-06-15 10:47:39.815223028 +0800
+++ b/dlls/wineoss.drv/mmdevdrv.c	2017-06-15 10:51:51.531212501 +0800
@@ -56,8 +56,8 @@
 
 #define NULL_PTR_ERR MAKE_HRESULT(SEVERITY_ERROR, FACILITY_WIN32, RPC_X_NULL_REF_POINTER)
 
-static const REFERENCE_TIME DefaultPeriod = 100000;
-static const REFERENCE_TIME MinimumPeriod = 50000;
+static const REFERENCE_TIME DefaultPeriod = 20000;
+static const REFERENCE_TIME MinimumPeriod = 10000;
 
 struct ACImpl;
 typedef struct ACImpl ACImpl;
diff -ur a/dlls/winepulse.drv/pulse.c b/dlls/winepulse.drv/pulse.c
--- a/dlls/winepulse.drv/pulse.c	2021-07-31 03:19:29.000000000 +0800
+++ b/dlls/winepulse.drv/pulse.c	2021-08-14 20:41:48.766707810 +0800
@@ -93,8 +93,8 @@
 
 static UINT g_phys_speakers_mask = 0;
 
-static const REFERENCE_TIME MinimumPeriod = 30000;
-static const REFERENCE_TIME DefaultPeriod = 100000;
+static const REFERENCE_TIME MinimumPeriod = 10000;
+static const REFERENCE_TIME DefaultPeriod = 20000;
 
 static pthread_mutex_t pulse_mutex;
 static pthread_cond_t pulse_cond = PTHREAD_COND_INITIALIZER;
