diff -Nurp a/configure b/configure
--- a/configure	2021-05-29 02:49:43.000000000 +0800
+++ b/configure	2021-08-31 00:21:52.421898073 +0800
@@ -46399,6 +46399,7 @@ else $as_nop
       i?86 | x86_64 | \
       i860 | \
       ia64 | \
+      loongarch* | \
       m32r | \
       m68* | \
       m88k | \
diff -Nurp a/m4/stack-direction.m4 b/m4/stack-direction.m4
--- a/m4/stack-direction.m4	2021-05-19 22:35:46.000000000 +0800
+++ b/m4/stack-direction.m4	2021-08-31 00:22:06.736770510 +0800
@@ -31,6 +31,7 @@ AC_DEFUN([SV_STACK_DIRECTION],
       i?86 | x86_64 | \
       i860 | \
       ia64 | \
+      loongarch* | \
       m32r | \
       m68* | \
       m88k | \
