From c2e7b8b78f1eb37559b13b4fc0c9919c904c4637 Mon Sep 17 00:00:00 2001
From: WANG Xuerui <git@xen0n.name>
Date: Sun, 8 Aug 2021 12:13:16 +0800
Subject: [PATCH 09/11] LoongArch: add "ret" instruction to reduce typing

---
 opcodes/loongarch-opc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/opcodes/loongarch-opc.c b/opcodes/loongarch-opc.c
index fd3dc55707c..e05bbfd1817 100644
--- a/opcodes/loongarch-opc.c
+++ b/opcodes/loongarch-opc.c
@@ -757,6 +757,7 @@ static struct loongarch_opcode loongarch_jmp_opcodes[] =
   { 0, 0, "bltu", "r,r,la", "bleu %2,%1,%%pcrel(%3)", 0, 0, 0 },
   { 0, 0, "bgeu", "r,r,la", "bgtu %2,%1,%%pcrel(%3)", 0, 0, 0 },
   { 0, 0, "jr", "r", "jirl $r0,%1,0", 0, 0, 0 },
+  { 0, 0, "ret", "", "jirl $r0,$r1,0", 0, 0, 0 },
 
   { 0 } /* Terminate the list.  */
 };
-- 
2.30.1

