From 66362e81facef7478667f8bc76e24bf37f0f84bc Mon Sep 17 00:00:00 2001
From: Calring <519384449@qq.com>
Date: Wed, 11 Aug 2021 11:21:46 +0800
Subject: [PATCH 11/30] Add some annotation of instruction opcodes.

---
 gas/config/tc-loongarch.c | 17 ++++++++---------
 1 file changed, 8 insertions(+), 9 deletions(-)

diff --git a/gas/config/tc-loongarch.c b/gas/config/tc-loongarch.c
index 8543a2f27c2..fcb223a14c0 100644
--- a/gas/config/tc-loongarch.c
+++ b/gas/config/tc-loongarch.c
@@ -656,9 +656,9 @@ check_this_insn_before_appending (struct loongarch_cl_insn *ip)
       ip->reloc_num++;
     }
   else if (ip->insn->mask == 0xffff8000 &&
-           ((ip->insn_bin & 0xfff00000) == 0x38600000 ||
-            (ip->insn_bin & 0xffff0000) == 0x38700000 ||
-            (ip->insn_bin & 0xffff0000) == 0x38710000))
+           ((ip->insn_bin & 0xfff00000) == 0x38600000 ||  /* amswap.w  rd, rk, rj */
+            (ip->insn_bin & 0xffff0000) == 0x38700000 ||  /* ammax_db.wu  rd, rk, rj */
+            (ip->insn_bin & 0xffff0000) == 0x38710000))  /* ammin_db.wu  rd, rk, rj */
     {
       /* For AMO insn amswap.[wd], amadd.[wd], etc. */
       if (ip->args[0] != 0 &&
@@ -667,18 +667,17 @@ check_this_insn_before_appending (struct loongarch_cl_insn *ip)
           _ ("AMO insns require rd != base && rd != rt when rd isn't $r0"));
     }
   else if ((ip->insn->mask == 0xffe08000 &&
-            (ip->insn_bin & 0xffe00000) == 0x00600000) ||
+            (ip->insn_bin & 0xffe00000) == 0x00600000) ||  /* bstrins.w  rd, rj, msbw, lsbw */
            (ip->insn->mask == 0xffc00000 &&
-            (ip->insn_bin & 0xff800000) == 0x00800000))
+            (ip->insn_bin & 0xff800000) == 0x00800000))  /* bstrins.d  rd, rj, msbd, lsbd */
     {
       /* For bstr(ins|pick).[wd] */
       if (ip->args[2] < ip->args[3])
         as_fatal (_ ("bstr(ins|pick).[wd] require msbd >= lsbd"));
     }
-  else if (ip->insn->mask != 0 && (ip->insn_bin & 0xfe0003c0) == 0x04000000 &&
-           (strcmp ("csrxchg", ip->name) == 0 ||
-            strcmp ("gcsrxchg", ip->name) == 0))
-    as_fatal (_ ("g?csrxchg require rj != $r0 && rj != $r1"));
+  else if (ip->insn->mask != 0 && (ip->insn_bin & 0xfe0003c0) == 0x04000000 && /*  csrxchg  rd, rj, csr_num */
+           (strcmp ("csrxchg", ip->name) == 0))
+    as_fatal (_ ("csrxchg require rj != $r0 && rj != $r1"));
 
   return ret;
 }
-- 
2.30.1

