From f67b94fe59c57d8c3737b008b59ddafc5dba4fbf Mon Sep 17 00:00:00 2001
From: liuzhensong <liuzhensong@loongson.cn>
Date: Thu, 12 Aug 2021 20:22:38 +0800
Subject: [PATCH 22/30] bfd: delete jr insn

  bfd/elfnn-loongarch.c
---
 bfd/elfnn-loongarch.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/bfd/elfnn-loongarch.c b/bfd/elfnn-loongarch.c
index 9fe8d9b4ffc..61518e37d56 100644
--- a/bfd/elfnn-loongarch.c
+++ b/bfd/elfnn-loongarch.c
@@ -198,8 +198,6 @@ loongarch_make_plt_entry (bfd_vma got_plt_entry_addr, bfd_vma plt_entry_addr,
 	     | (lo & 0xfff) << 10;
   entry[2] = 0x4c0001ed;	/* jirl $r13, $15, 0 */
   entry[3] = 0x03400000;	/* nop */
-  /* entry[2] = 0x1c00000d; */ /* pcaddu12i $13, 4 */
-  /* entry[3] = 0x4c0001e0; */ /* jirl $r0, $15, 0 */
 }
 
 /* Create an entry in an LoongArch ELF linker hash table.  */
-- 
2.30.1

