From 6c4044c7b7b5cbbb3390807dc7a3a4bd2bf5cdb4 Mon Sep 17 00:00:00 2001
From: WANG Xuerui <git@xen0n.name>
Date: Sun, 8 Aug 2021 12:13:16 +0800
Subject: [PATCH 27/30] opcodes: LoongArch: add "ret" instruction to reduce
 typing

---
 opcodes/loongarch-opc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/opcodes/loongarch-opc.c b/opcodes/loongarch-opc.c
index 03dc18f3add..74ea32f4552 100644
--- a/opcodes/loongarch-opc.c
+++ b/opcodes/loongarch-opc.c
@@ -759,6 +759,7 @@ static struct loongarch_opcode loongarch_jmp_opcodes[] =
   { 0x0,	0x0,		"bltu",		"r,r,la",			"bgtu %2,%1,%%pcrel(%3)",	0, 0, 0 },
   { 0x0,	0x0,		"bgeu",		"r,r,la",			"bleu %2,%1,%%pcrel(%3)",	0, 0, 0 },
   { 0x0,	0x0,		"jr",		"r",				"jirl $r0,%1,0",		0, 0, 0 },
+  { 0x0,	0x0,		"ret",		"",				"jirl $r0,$r1,0",		0, 0, 0 },
 
   { 0 } /* Terminate the list.  */
 };
-- 
2.30.1

