From 8504a4c52f1e4885974ca81b767045771f8a9112 Mon Sep 17 00:00:00 2001
From: WANG Xuerui <git@xen0n.name>
Date: Fri, 18 Jun 2021 17:26:56 +0800
Subject: [PATCH 09/10] LoongArch: fix assembler TLS support detection

---
 gcc/configure.ac | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/gcc/configure.ac b/gcc/configure.ac
index 28089085337..8237fedd63f 100644
--- a/gcc/configure.ac
+++ b/gcc/configure.ac
@@ -3822,6 +3822,17 @@ foo:	data8	25
 	tls_first_minor=13
 	tls_as_opt=--fatal-warnings
 	;;
+  loongarch*-*-*)
+    conftest_s='
+	.section .tdata,"awT",@progbits
+x:	.word 2
+	.text
+	la.tls.gd $a0,x
+	bl __tls_get_addr'
+	tls_first_major=0
+	tls_first_minor=0
+	tls_as_opt='--fatal-warnings'
+	;;
   microblaze*-*-*)
     conftest_s='
 	.section .tdata,"awT",@progbits
-- 
2.30.1

