From 6c0ab26d680f08f245e3a51743c76a9627e9d6fe Mon Sep 17 00:00:00 2001
From: Xiaotian Wu <wuxiaotian@loongson.cn>
Date: Sat, 24 Jul 2021 18:15:07 +0800
Subject: [PATCH] use interpreter /lib/ld-linux-loongarch{32,64}.so.1

---
 gcc/config/loongarch/linux.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gcc/config/loongarch/linux.h b/gcc/config/loongarch/linux.h
index 6b1d837e809..f71e3c9600d 100644
--- a/gcc/config/loongarch/linux.h
+++ b/gcc/config/loongarch/linux.h
@@ -30,9 +30,9 @@ along with GCC; see the file COPYING3.  If not see
 #define GNU_USER_LINK_EMULATION64 "elf64loongarch"
 
 #define GLIBC_DYNAMIC_LINKERLP32 \
-  "/lib32/ld.so.1"
+  "/lib32/ld-linux-loongarch32.so.1"
 #define GLIBC_DYNAMIC_LINKERLP64 \
-  "/lib64/ld.so.1"
+  "/lib64/ld-linux-loongarch64.so.1"
 
 #define GNU_USER_DYNAMIC_LINKERLP32 GLIBC_DYNAMIC_LINKERLP32
 #define GNU_USER_DYNAMIC_LINKERLP64 GLIBC_DYNAMIC_LINKERLP64
-- 
2.30.1

