From d8eed345be10c4a7af53c6ab993b6c77eae26e85 Mon Sep 17 00:00:00 2001
From: Xiaotian Wu <wuxiaotian@loongson.cn>
Date: Sat, 24 Jul 2021 18:15:07 +0800
Subject: [PATCH 07/10] use interpreter /libNN/ld-linux-loongarchNN.so.1

---
 gcc/config/loongarch/linux.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gcc/config/loongarch/linux.h b/gcc/config/loongarch/linux.h
index 56da293bd11..3ee0671367e 100644
--- a/gcc/config/loongarch/linux.h
+++ b/gcc/config/loongarch/linux.h
@@ -31,8 +31,8 @@ along with GCC; see the file COPYING3.  If not see
 #define GNU_USER_LINK_EMULATION32 "elf32loongarch"
 #define GNU_USER_LINK_EMULATION64 "elf64loongarch"
 
-#define GLIBC_DYNAMIC_LINKERLP32 "/lib32/ld.so.1"
-#define GLIBC_DYNAMIC_LINKERLP64 "/lib64/ld.so.1"
+#define GLIBC_DYNAMIC_LINKERLP32 "/lib32/ld-linux-loongarch32.so.1"
+#define GLIBC_DYNAMIC_LINKERLP64 "/lib64/ld-linux-loongarch64.so.1"
 
 #define GNU_USER_DYNAMIC_LINKERLP32 GLIBC_DYNAMIC_LINKERLP32
 #define GNU_USER_DYNAMIC_LINKERLP64 GLIBC_DYNAMIC_LINKERLP64
-- 
2.33.0

